common/lib/src/realm_types.dart

```diff
@@ -52,14 +52,23 @@ enum RealmCollectionType {
   dictionary,
 }
 
+class RealmError extends Error {}
+
+class RealmUnsupportedError extends UnsupportedError implements RealmError {
```

Comment:
I wonder do we gain much by extending errors here. Since these are not supposed to be catched by user code? Wouldn't a simple `RealmUnsupportedError ` throwing the relevant message suffice here. I expect string constants are interned in Dart

Reply:
They are, I just like to have a type hierarchy in exceptions, if you need to catch them.

Reply:
Fully agree for exceptions, but for Errors seems unnecessary since these should fail the app and are not supposed to be catched.

Reply:
I still like the distinction during development / debugging. You could argue the same for all the different types of error in dart. Why not just have one Error class?

Reply:
I reduced the hierarchy by one level